helper/backoff.go

```diff
@@ -29,3 +31,34 @@ func Backoff(backoffBase time.Duration, backoffLimit time.Duration, attempt uint
 
 	return deadline
 }
+
+// WithBackoffFunc is a helper that runs a function with geometric backoff + a
+// small jitter to a maximum backoff. It returns once the context closes, with
+// the error wrapping over the error from the function.
```

Comment:
Note for reviewers: this logic was lifted from the `csi_hook`, which is reasonably battle-tested at this point. Once we merge this I'll go back and update the `csi_hook` to use this function.